pkg/route/apis/route/v1/conversion_test.go

```diff
@@ -41,3 +50,108 @@ func TestSupportingCamelConstants(t *testing.T) {
 		}
 	}
 }
+
+func TestRouteSpecConversion(t *testing.T) {
+	scheme := runtime.NewScheme()
+	Install(scheme)
+
+	serviceName := "TestService"
+	serviceWeight := int32(0)
+	var headerNameXFrame string = "X-Frame-Options"
+
+	testcases := map[string]struct {
+		versionedRouteSpec1   *v1.RouteSpec
+		unversionedRouteSpec2 *route.RouteSpec
+	}{
+		"RouteSpec Conversion 1": {
+			versionedRouteSpec1: &v1.RouteSpec{
+				Host: "host",
+				Path: "path",
+				Port: &v1.RoutePort{
+					TargetPort: intstr.FromInt(8080),
+				},
+				To: v1.RouteTargetReference{
+					Name:   serviceName,
+					Weight: &serviceWeight,
+				},
+				TLS: &v1.TLSConfig{
+					Termination:              v1.TLSTerminationEdge,
+					Certificate:              "abc",
+					Key:                      "def",
+					CACertificate:            "ghi",
+					DestinationCACertificate: "jkl",
+				},
+				HTTPHeaders: &v1.RouteHTTPHeaders{
+
+					Actions: v1.RouteHTTPHeaderActions{
+						Response: []v1.RouteHTTPHeader{
+							{
+								Name: headerNameXFrame,
+								Action: v1.RouteHTTPHeaderActionUnion{
+									Type: v1.Set,
+									Set: &v1.RouteSetHTTPHeader{
+										Value: "DENY",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			unversionedRouteSpec2: &route.RouteSpec{
+				Host: "host",
+				Path: "path",
+				Port: &route.RoutePort{
+					TargetPort: intstr.FromInt(8080),
+				},
+				To: route.RouteTargetReference{
+					Name:   serviceName,
+					Weight: &serviceWeight,
+				},
+				TLS: &route.TLSConfig{
+					Termination:              route.TLSTerminationEdge,
+					Certificate:              "abc",
+					Key:                      "def",
+					CACertificate:            "ghi",
+					DestinationCACertificate: "jkl",
+				},
+				HTTPHeaders: &route.RouteHTTPHeaders{
+
+					Actions: route.RouteHTTPHeaderActions{
+						Response: []route.RouteHTTPHeader{
+							{
+								Name: headerNameXFrame,
+								Action: route.RouteHTTPHeaderActionUnion{
+									Type: route.Set,
+									Set: &route.RouteSetHTTPHeader{
+										Value: "DENY",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for k, tc := range testcases {
+		// un-versioned -> versioned
+		internal1 := &v1.RouteSpec{}
+		if err := Convert(tc.unversionedRouteSpec2, internal1, nil); err != nil {
+			t.Errorf("%q - %q: unexpected error: %v", k, "from unversioned to versioned", err)
```

Comment:
I suppose for consistency with the other errors in this test and with the examples that p0lyn0mial referenced, this should be the following:
```suggestion
			t.Errorf("%q - %q: unexpected error: %v", k, "from route to routev1", err)

Reply:
Yes made this change